src/java/main/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1041,6 +1041,8 @@ private void sendPing() {
 
         private InetSocketAddress rwServerAddress = null;
 
+        private InetSocketAddress addr = null;
```

Comment:
[non-blocking]

The name could be an ambiguous: is this the address of the local (client) side of the connection, or the remote (server) side of the connection?  Especially as there is `rwServerAddress` field member defined immediately above.

Suggest `serverAddress` (or similar)

Reply:
Remote side of the connection. Basically the ip address that the client is trying to connect to. ```rwServerAddress``` is the address of non-R/O server found if there's any.

I'll change this to ```serverAddress```.